src/main/java/com/dataloom/authorization/Principals.java

```diff
@@ -30,7 +30,7 @@ public static Principal getCurrentUser() {
         return new Principal(
                 PrincipalType.USER,
                 ( (Auth0UserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal() )
-                        .getUsername() );
+                        .getAuth0Attribute("user_id").toString() );
```

Comment:
I ran it locally :p. This code is running against the standard auth0 integration, and the auth0 docs guarantee this property, but they have been wrong before. 

We definitely want to move to this - username is volatile and can be pretty easily faked. Auth0UserDetails tries to fill the username field with a lot of different properties, starting with `email`. That leaves accounts that have been created with username/pass open to an attack where a malicious user creates a FB account with the victims email, then signs in using FB.